sources/api/early-boot-config/src/main.rs

```diff
@@ -31,6 +30,8 @@ use crate::provider::{Platform, PlatformDataProvider};
 // ConditionPathExists that will prevent it from running again if this file exists.
 // We create it after running successfully.
 const MARKER_FILE: &str = "/var/lib/bottlerocket/early-boot-config.ran";
+/// The directory containing user data provider binaries
+const PROVIDERS_DIR: &str = "/usr/share/early-boot-config/data-providers.d";
```

Comment:
This will affect the packaging, but I'd recommend this for binaries:
```suggestion
/// The directory containing user data provider binaries
const PROVIDERS_DIR: &str = "/usr/libexec/early-boot-config/data-providers.d";
```

It's slightly weird to put binaries in `/usr/share` - the [FHS](https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html#usrshareArchitectureindependentData) says that's for "architecture-independent data".

(This wouldn't Bottlerocket's only crime against the FHS, but still.)

(Need to fix the docstring if you update this.)